Modules/Registration/RegistrationMethodsv4/include/itkImageRegistrationMethodv4.hxx

```diff
@@ -99,8 +99,7 @@ ImageRegistrationMethodv4<TFixedImage, TMovingImage, TTransform, TVirtualImage,
   this->ProcessObject::SetNthOutput(0, transformDecorator);
   this->m_OutputTransform = transformDecorator->GetModifiable();
 
-  // By default we set up a 3-level image registration.
-  this->SetNumberOfLevels(3);
+  this->SetNumberOfLevels(m_NumberOfLevels);
```

Comment:
Please consider removing the `this->SetNumberOfLevels` call.  The function call seems completely redundant now that `this->m_NumberOfLevels` is already initialized by the in-class `{ 3 }` default member initialization.

Reply:
@N-Dekker please have a closer look at the `SetNumberOfLevels` method implementation.

Reply:
It is guarded by `if (this->m_NumberOfLevels != numberOfLevels)`, which makes this call redundant? If we want this to be called, it would make sense to make in-class initialization to 0.

Reply:
Uops you're right. Thanks for the heads-up @dzenanz ! Amended the commit to initialize it to 0 in 54367db.

Reply:
Thanks for your effort so far, Jon. Do I understand correctly that this PR now changes the NumberOfLevels of a default constructed ImageRegistrationMethodv4 from `3` to `0`? If so, is that intended?

Reply:
:sweat_smile: again my mistake :hand:. Thanks for the head-up @N-Dekker. Fixed in e939dab. I think the initialization needs to be reworked to make it more consistent, as the `m_SmoothingSigmasPerLevel` and the `m_MetricSamplingPercentagePerLevel` ivars are initialized both in the constructor and the method at issue here to different values. But I'd leave it for a separate PR.

Reply:
> I think the initialization needs to be reworked to make it more consistent, as the m_SmoothingSigmasPerLevel and the m_MetricSamplingPercentagePerLevel ivars are initialized both in the constructor and the method at issue here to different values. But I'd leave it for a separate PR.

PR https://github.com/InsightSoftwareConsortium/ITK/pull/3876.